src/SprykerSdk/Zed/AppSdk/Business/ReadinessChecker/Checker/ComposerChecker.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Zed\AppSdk\Business\ReadinessChecker\Checker;
+
+use Composer\InstalledVersions;
+use Composer\Semver\VersionParser;
+use Generated\Shared\Transfer\CheckConfigurationTransfer;
+use Generated\Shared\Transfer\CheckerMessageTransfer;
+use Generated\Shared\Transfer\RecipeTransfer;
+
+class ComposerChecker implements CheckerInterface
+{
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return 'composer';
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\RecipeTransfer $recipeTransfer
+     * @param \Generated\Shared\Transfer\CheckConfigurationTransfer $checkConfigurationTransfer
+     *
+     * @return \Generated\Shared\Transfer\RecipeTransfer
+     */
+    public function check(RecipeTransfer $recipeTransfer, CheckConfigurationTransfer $checkConfigurationTransfer): RecipeTransfer
```

Comment:
why checker returns what it checks ?

Reply:
Because it contains the validation error specific for this Recipe.